app/services/SubscribeService.scala

```diff
@@ -106,16 +106,42 @@ trait SubscribeService {
   private def createEnrolmentVerifiers(response: HttpResponse, data: JsValue): Verifiers = {
     getUtrAndPostCode(data) match {
       case (Some(uniqueTaxRef), Some(ukClientPostCode)) =>
-        Verifiers(List(
-          Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
-          Verifier(GovernmentGatewayConstants.VerifierCtUtr, uniqueTaxRef))
-        )
+        (data \ "businessType").asOpt[String] match {
+          case Some("LLP") => {
+            Verifiers(List(
+              Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
+              Verifier(GovernmentGatewayConstants.VerifierSaUtr, uniqueTaxRef))
+            )
+          }
+          case Some("Corporate Body") => {
+            Verifiers(List(
+              Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
+              Verifier(GovernmentGatewayConstants.VerifierCtUtr, uniqueTaxRef))
+            )
+          }
+          case Some("Partnership") => {
+            Verifiers(List(
+              Verifier(GovernmentGatewayConstants.VerifierPostalCode, ukClientPostCode),
+              Verifier(GovernmentGatewayConstants.VerifierSaUtr, uniqueTaxRef))
+            )
+          }
+        }
       case (None, Some(nonUkClientPostCode)) =>
         Verifiers(List(
           Verifier(GovernmentGatewayConstants.VerifierNonUKPostalCode, nonUkClientPostCode))
         ) //N.B. Non-UK Clients might use the property UK Postcode or their own Non-UK Postal Code
       case (Some(uniqueTaxRef), None) =>
-        Verifiers(List(Verifier(GovernmentGatewayConstants.VerifierCtUtr, uniqueTaxRef)))
+        (data \ "businessType").asOpt[String] match {
+          case Some("LLP") => {
+            Verifiers(List(Verifier(GovernmentGatewayConstants.VerifierSaUtr, uniqueTaxRef)))
+          }
+          case Some("Corporate Body") => {
```

Comment:
Can you ensure Corporate Body and Partnership are tested please?

(these and the above are red on coverage)

Reply:
I ran coverage on the new code that's been pushed up and they are now tested